modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/ViewSharedWithMeSectionDisplayContext.java

```diff
@@ -18,14 +25,38 @@
 public class ViewSharedWithMeSectionDisplayContext {
 
 	public ViewSharedWithMeSectionDisplayContext(
+		DepotEntryLocalService depotEntryLocalService,
 		HttpServletRequest httpServletRequest) {
 
+		_depotEntryLocalService = depotEntryLocalService;
 		_httpServletRequest = httpServletRequest;
 	}
 
 	public String getAPIURL() {
-		return "/o/headless-admin-user/v1.0/my-user-account/shared-assets" +
-			"/shared-with-me";
+		StringBundler sb = new StringBundler(4);
+
+		sb.append(
+			"/o/headless-admin-user/v1.0/my-user-account/shared-assets"
+		).append(
+			"/shared-with-me?emptySearch=true"
+		);
+
+		List<String> groupIds = TransformUtil.transform(
+			_depotEntryLocalService.getDepotEntries(
+				QueryUtil.ALL_POS, QueryUtil.ALL_POS),
+			depotEntry -> String.valueOf(depotEntry.getGroupId()));
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)_httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		groupIds.add(String.valueOf(themeDisplay.getScopeGroupId()));
+
+		sb.append("&filter=groupId in ('");
+		sb.append(String.join("','", groupIds));
+		sb.append("')");
+
+		return sb.toString();
```

Comment:
This will cause problems if there are many asset libraries. I think it would be better to add a new query parameter to the resource (e.g. /my-user-account/shared-assets/shared-with-me?emptySearch=true&cms=true), and change the shared entry indexer to add that property if it belongs to a space.

Reply:
At this moment, Can we differ between an asset lib and a space. because I think that the answer is no.
correct me if I'm wrong.
So at this moment, adding a new filter has no result.
Once we can make a different thing, we can change the _depotEntryLocalService.getDepotEntries( to the new one.

Reply:
Ok, I will rewrite all

Reply:
done, please, re -review :) 

naming .. well :D

Reply:
I have spoken with Alicia, what do you think to use the filter in order to add the  the cms "parameter" on it @adolfopa ? or do tou think it is better to have a real parameter in the headless api? I open for anything but I think that it is better for the moment to have as a filter since for me it it strange to have a parameter called cms